### PR TITLE
Admin action for listing all clusters

### PR DESCRIPTION
**Pull request by ehashman**
See [Admin API list by region](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/6098213).

Based on the public list clusters by subscription API call, but with a few differences because YAGNI?:

- no enrichment of the data for now
- no pagination for now (if we need that, unclear if we should be paginating here or on the admin actions backend)

Will attempt to write the backend C# code next so I can test e2e.

### CODE REVIEW

File: pkg/frontend/admin_openshiftcluster_list.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+func (f *frontend) _getAdminOpenShiftClusters(ctx context.Context, r *http.Request, converter api.OpenShiftClusterConverter) ([]byte, error) {
+	i, err := f.db.OpenShiftClusters.List()
+	if err != nil {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (jim-minter) [inline pkg/frontend/admin_openshiftcluster_list.go:36]: dupe

File: pkg/frontend/admin_openshiftcluster_list.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			ocs = append(ocs, doc.OpenShiftCluster)
+		}
+	}
+
+	for i := range ocs {
+		ocs[i].Properties.ClusterProfile.PullSecret = ""
+		ocs[i].Properties.ServicePrincipalProfile.ClientSecret = ""
+	}
+
+	// NOTE(ehashman): sort of a hack, must currently provide a nextPage link, so I left it blank
```
Comment (jim-minter) [inline pkg/frontend/admin_openshiftcluster_list.go:59]: seems ok to me

File: pkg/frontend/admin_openshiftcluster_list.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+func (f *frontend) getAdminOpenShiftClusters(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	b, err := f._getAdminOpenShiftClusters(ctx, r, f.apis["admin"].OpenShiftClusterConverter())
+
+	reply(log, w, nil, b, err)
```
Comment (jim-minter) [inline pkg/frontend/admin_openshiftcluster_list.go:25]: should be adminReply, change might affect your unit tests

File: pkg/frontend/admin_openshiftcluster_list.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getAdminOpenShiftClusters(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	b, err := f._getAdminOpenShiftClusters(ctx, r, f.apis["admin"].OpenShiftClusterConverter())
```
Comment (jim-minter) [inline pkg/frontend/admin_openshiftcluster_list.go:23]: s/"admin"/admin.APIVersion/

File: pkg/database/openshiftclusters.go [hunk 5]
```diff
@@ -220,6 +221,10 @@ func (c *openShiftClusters) ChangeFeed() cosmosdb.OpenShiftClusterDocumentIterat
 	return c.c.ChangeFeed(nil)
 }
 
+func (c *openShiftClusters) List() (cosmosdb.OpenShiftClusterDocumentIterator, error) {
```
Comment (jim-minter) [inline pkg/database/openshiftclusters.go:224]: should be `func (c *openShiftClusters) List() cosmosdb.OpenShiftClusterDocumentIterator`

**Comment by jim-minter** [comment]
one more nit, sorry

**Comment by ehashman** [comment]
@jim-minter should be gtg

**Pull request merged**